core/src/main/scala/com/avast/grpc/jsonbridge/GrpcJsonBridge.scala

```diff
@@ -1,31 +1,24 @@
 package com.avast.grpc.jsonbridge
 
-import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcHeader
-import io.grpc.{BindableService, Status}
+import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcMethodName
+import io.grpc.Status
 
 import scala.language.higherKinds
 
-trait GrpcJsonBridge[F[_], Service <: BindableService] extends AutoCloseable {
-
-  /** Invokes method with given name, if it exists. The method should never return a failed Future.
-    *
-    * @param name Name of the method.
-    * @param json Method input (JSON string).
-    * @return Left if there was some error, Right otherwise.
-    */
-  def invokeGrpcMethod(name: String, json: => String, headers: => Seq[GrpcHeader] = Seq.empty): F[Either[Status, String]]
-
-  /** Returns sequence of method names supported by this `GrpcJsonBridge`.
-    */
-  def methodsNames: Seq[String]
-
-  /** Returns name of the service (FQN).
-    */
-  def serviceName: String
+trait GrpcJsonBridge[F[_]] {
+  def invoke(methodName: GrpcMethodName, body: String, headers: Map[String, String]): F[Either[Status, String]]
+  def methodsNames: Seq[GrpcMethodName]
+  def servicesNames: Seq[String]
 }
 
 object GrpcJsonBridge {
-
-  case class GrpcHeader(name: String, value: String)
-
+  case class GrpcMethodName(service: String, method: String) {
+    val fullName: String = service + "/" + method
+  }
+  object GrpcMethodName {
+    def apply(fullMethodName: String): GrpcMethodName = {
+      val s = fullMethodName.split('/')
```

Comment:
not the most important point, but still worth rewriting:
```
val Seq(service, method) = fullMethodName.split('/')
```
this will throw in case it has more than 2 elems (which is what we want)